test/helpers/PlayerInteractionWrapper.js

```diff
@@ -309,6 +317,15 @@ class PlayerInteractionWrapper {
         });
     }
 
+    attachOpponentOwnedUpgrades(opponentOwnedUpgrades = []) {
+        opponentOwnedUpgrades.forEach((upgrade) => {
+            const upgradeCard = this.findCardByName(upgrade.card, 'any', 'opponent');
+            const attachedCardAlsoOpponentControlled = upgrade.hasOwnProperty('attachedToOwner') && upgrade.attachedToOwner !== this.player.nameField;
```

Comment:
Having trouble following - is the `attachedToOwner` property getting added to the card object itself, or is this some other wrapper object or something? I'd like to avoid modifying the card properties if possible

Reply:
'upgrade' is the json version, 'upgradeCard' is the actual card